framework/src/problems/FEProblemBase.C

```diff
@@ -7262,6 +7263,12 @@ FEProblemBase::checkNonlinearConvergence(std::string & msg,
   TIME_SECTION("checkNonlinearConvergence", 5, "Checking Nonlinear Convergence");
   mooseAssert(_current_nl_sys, "This should be non-null");
 
+  if (_fail_next_nonlinear_convergence_check)
+  {
+    _fail_next_nonlinear_convergence_check = false;
```

Comment:
Are both resets necessary? This is not all that important of a question

Reply:
probably not. let me double check...

Reply:
Hm, not clear to me. Two scenarios come to mind

* anyone overrides the `checkNonlinearConvergence` function - we'd need the reset in `solve` for MOOSE to still work
* anything relied on evaluating the residual between the last  `checkNonlinearConvergence` and the next call to `solve` (rescaling?)

Having two resets would be a bit more robust.

Reply:
Yea this is fine